test/ConnectionFactoryTest.php

```diff
@@ -119,7 +118,31 @@ public function testParamsInjection(): void
             'username' => 'foo',
             'driverClass' => PDOSqliteDriver::class,
             'wrapperClass' => null,
-            'pdo' => null,
+        ], $connection->getParams());
+    }
+
+    public function testUrlParam(): void
+    {
+        $factory    = new ConnectionFactory();
+        $connection = $factory($this->buildContainer('orm_default', 'orm_default', [
+            'params' => [
+                'url' => 'sqlite3:///:memory:',
+                'primary' => ['url' => 'pdo-mysql://localhost:4486/foo?charset=utf8mb4'],
```

Comment:
I would like the test to represent the reality: `params.url` shouldn't go along with `params.primary`, right?
Can we somehow warn the user about the misuage?

Reply:
AFAIK, `primary`/`replica` parameters are used when the `wrapperClass` set to `Doctrine\DBAL\Connections\PrimaryReadReplicaConnection`. In this case all the root params are ignored, expect the `driver` and `keepReplica` params. And that will be enough.

Reply:
Yeah, is like the tests to reflect the real use case please

Reply:
Do you want to split into two tests, one for `params.url` and the other for `params.primary.url`/`params.replica.url`?

Reply:
> Yeah, is like the tests to reflect the real use case please

I've added a separate test for primary-replica setup.